var/spack/repos/builtin/packages/prmon/package.py

```diff
@@ -17,6 +17,7 @@ class Prmon(CMakePackage):
     maintainers = ['vvolkl']
 
     version("master", branch="master")
+    version('2.2.0', sha256='e11ee9a5ed7d007a064b18bd75f799a8055b8dc6da241d97a980a1e13c4a2495')
```

Comment:
Very strange.  This is *not* the value I get running `spack checksum prmon@2.2.0`.  I get `7fcdf6ac7284ad38b496e8487ccb861bee5f58bd232282009fde459c751c177c`.

How did you calculate the `sha256`?

Reply:
I should note the tarball used in the above command is `https://github.com/HSF/prmon/archive/v2.2.0.tar.gz`.

Reply:
@tldahlgren thanks for checking that!  Indeed it seems that spack checksum uses the binary asset of the release by error:

 `2.2.0  https://github.com/HSF/prmon/releases/download/v2.2.0/prmon_2.2.0_x86_64-static-gnu93-opt.tar.gz`

I'll fix that and check why it hit that url.

Reply:
Interesting.  Spack performs a version substitution on the `url` for packages like this; hence the URL it used for me in my comment above.

Reply:
hmmm, which spack version are you using? I am on `develop`, basically:

```
~/repo/spack (develop)$ spack debug report
* **Spack:** 0.16.1-2348-c3733771e4
* **Python:** 3.8.5
* **Platform:** linux-ubuntu20.04-skylake
* **Concretizer:** original
```

and I get:

```
~/repo/spack (develop)$ spack checksum prmon
==> Found 10 versions of prmon:
  
  2.2.0  https://github.com/HSF/prmon/releases/download/v2.2.0/prmon_2.2.0_x86_64-static-gnu93-opt.tar.gz
  2.1.1  https://github.com/HSF/prmon/releases/download/v2.1.1/prmon_2.1.1_x86_64-static-gnu93-opt.tar.gz
  2.1.0  https://github.com/HSF/prmon/releases/download/v2.1.0/prmon_2.1.0_x86_64-static-gnu93-opt.tar.gz
  2.0.2  https://github.com/HSF/prmon/releases/download/v2.0.2/prmon_2.0.2_x86_64-static-gnu93-opt.tar.gz
  2.0.1  https://github.com/HSF/prmon/releases/download/v2.0.1/prmon_2.0.1_x86_64-static-gnu93-opt.tar.gz
  2.0.0  https://github.com/HSF/prmon/releases/download/v2.0.0/prmon_2.0.0_x86_64-static-gnu93-opt.tar.gz
  1.1.1  https://github.com/HSF/prmon/releases/download/v1.1.1/prmon_1.1.1_x86_64-static-gnu73-opt.tar.gz
  1.1.0  https://github.com/HSF/prmon/archive/refs/tags/v1.1.0.tar.gz
  1.0.1  https://github.com/HSF/prmon/releases/download/v1.0.1/prmon_1.0.1_x86_64-static-gnu72-opt.tar.gz
  1.0.0  https://github.com/HSF/prmon/archive/refs/tags/v1.0.0.tar.gz

==> How many would you like to checksum? (default is 1, q to abort) 

```
the urls look very different, it seems to depend on how the assets were released?

Reply:
a very quick workaround is to change the url to use the `.zip` source archive instead of the `.tar.gz`, apparently then there is no more competition from the binary assets, which are only released  as `.tar.gz`
```
url      = "https://github.com/HSF/prmon/archive/refs/tags/v2.2.0.zip"

```


```
~/repo/spack (develop *)$ spack checksum prmon
==> Found 10 versions of prmon:
  
  2.2.0  https://github.com/HSF/prmon/archive/refs/tags/v2.2.0.zip
  2.1.1  https://github.com/HSF/prmon/archive/refs/tags/v2.1.1.zip
  2.1.0  https://github.com/HSF/prmon/archive/refs/tags/v2.1.0.zip
  2.0.2  https://github.com/HSF/prmon/archive/refs/tags/v2.0.2.zip
  2.0.1  https://github.com/HSF/prmon/archive/refs/tags/v2.0.1.zip
  2.0.0  https://github.com/HSF/prmon/archive/refs/tags/v2.0.0.zip
  1.1.1  https://github.com/HSF/prmon/archive/refs/tags/v1.1.1.zip
  1.1.0  https://github.com/HSF/prmon/archive/refs/tags/v1.1.0.zip
  1.0.1  https://github.com/HSF/prmon/archive/refs/tags/v1.0.1.zip
  1.0.0  https://github.com/HSF/prmon/archive/refs/tags/v1.0.0.zip
```

@tldahlgren would that be ok?

Reply:
> hmmm, which spack version are you using? I am on `develop`, basically:

oh I see, the different behavior is because I didn't specify the prmon version in the checksum command, never mind then!